### ISSUE TITLE
Respect order of directives against dynamic attributes

### ISSUE THREAD

**Issue by lxsmnsyc** [ryansolid/dom-expressions#103]
Currently, directives are always guaranteed to run first, regardless if the dynamic attribute is declared first.  We can consider this issue in the future however we can't do this right now considering that dom-expressions does not do true fine-grained attributes, as dynamic attributes are grouped into a single effect rather than their own independent effects. With ordered directives, it requires these dynamic attributes to either be split into their own independent effects or their group sequences (longest sequence of dynamic attributes before and after a directive, much like `mergeProps` is right now.), but of course we don't want this right now as this trades the creation/memory cost against the update cost.

**Comment by ryansolid** [comment]
Yeah it might be interesting adding a first run type mechanism. There is complexity there, but maybe a simple is first sort of check could allow us to run these things in order inside the effect if necessary. The cost of doing one extra check each time isn't going to sink us.

**Comment by LXSMNSYC** [comment]
Being optimistic, I'd say we could probably generate the effects first before the directives, so that the directives are guaranteed to run after all dynamic attributes has been set.

**Comment by ryansolid** [comment]
With ref/directive merge I think the current order still makes sense in the way that directives can produce effects so they shouldn't really be assumed to happen after.. I haven't had any reason to change the order yet. It is a debate I think we are pretty set in our current ordering unless there is a really compelling reason to change.
